### ISSUE TITLE
acc -> accuracy

### ISSUE THREAD

**Issue by jabowery** [abr/lmu#11]
In https://github.com/abr/neurips2019/blob/master/experiments/psMNIST-standard.ipynb I had to change:
```
plt.plot(result.history['val_acc'], label="Validation")
plt.plot(result.history['acc'], label="Training")
```
to
```
plt.plot(result.history['val_accuracy'], label="Validation")
plt.plot(result.history['accuracy'], label="Training")
```

This may be another versioning issue of some sort.

In order to get neurips2019 running at all on Windows 10, I had to perform some installation gymnastics.

**Comment by arvoelke** [comment]
Thanks I will look into this and see if there's a way I can set the versions in `setup.py` accordingly. Likely related to #10. Have you tried using Anaconda (https://www.anaconda.com/)? It's especially useful for managing Python environments on Windows, and so I'd recommend it in the future when installing large packages like TensorFlow / Keras / PyTorch.

**Comment by jabowery** [comment]
I always use `conda` and only use `pip` as a fallback.  However, since the instructions for installation were specific, I followed them.  I'll try setting up another `conda` environment and disaggregating `setup.py` into `conda` installs.

**Comment by arvoelke** [comment]
Oh I see. Yes I guess you would need to look inside of `setup.py` to see which conda installs you need to do in advance. Not too convenient I'll admit. There might be a way to make it convenient by deploying this on `conda-forge` but I have not looked into that yet. I may if this turns out to be a regular problem, or I might just decouple the requirements of the examples from the requirements of the cell itself (which should be compatible with a much larger set of versions). Let me know if you learn anything about certain versions needing to line up in a certain way as that would be a big help! Thanks again.

**Comment by jabowery** [comment]
The `acc` and `val_acc` references worked in a (min) `conda`environment under Windows 10 :
```
(base) > conda create -n lmuconda python=3.6
(base) > conda install 'tensorflow>=1.12.0,<2' keras scikit-learn matplotlib IPython notebook seaborn numpy scipy pytorch
(base) > conda activate lmuconda
(condalmu) > git clone --recurse-submodules https://github.com/abr/neurips2019 lmu
(condalmu) > cd lmu
```
but editing `setup.py` to simplify the `install_requires`:
```
      install_requires=[
          'nengolib>=0.5.1',
          'nengo>=2.8.0,<3.0',
          'phased-lstm-keras',
      ],
```
before executing the:
```
(condalmu) lmu>  pip install -e .
```
The output from `conda list`:
```
# Name                    Version                   Build  Channel
_pytorch_select           1.1.0                       cpu
_tflow_select             2.3.0                       mkl
absl-py                   0.8.1                    py36_0
astor                     0.8.0                    py36_0
attrs                     19.3.0                     py_0
backcall                  0.1.0                    py36_0
blas                      1.0                         mkl
bleach                    3.1.0                    py36_0
ca-certificates           2019.11.27                    0
certifi                   2019.11.28               py36_0
cffi                      1.13.2           py36h7a1dbc1_0
colorama                  0.4.1                    py36_0
cycler                    0.10.0           py36h009560c_0
decorator                 4.4.1                      py_0
defusedxml                0.6.0                      py_0
entrypoints               0.3                      py36_0
freetype                  2.9.1                ha9979f8_1
gast                      0.2.2                    py36_0
google-pasta              0.1.8                      py_0
grpcio                    1.16.1           py36h351948d_1
h5py                      2.9.0            py36h5e291fa_0
hdf5                      1.10.4               h7ebc959_0
icc_rt                    2019.0.0             h0cc432a_1
icu                       58.2                 ha66f8fd_1
importlib_metadata        1.2.0                    py36_0
intel-openmp              2019.4                      245
ipykernel                 5.1.3            py36h39e3cac_0
ipython                   7.10.1           py36h39e3cac_0
ipython_genutils          0.2.0                    py36_0
jedi                      0.15.1                   py36_0
jinja2                    2.10.3                     py_0
joblib                    0.14.0                     py_0
jpeg                      9b                   hb83a4c4_2
jsonschema                3.2.0                    py36_0
jupyter_client            5.3.4                    py36_0
jupyter_core              4.6.1                    py36_0
keras                     2.2.4                         0
keras-applications        1.0.8                      py_0
keras-base                2.2.4                    py36_0
keras-preprocessing       1.1.0                      py_1
kiwisolver                1.1.0            py36ha925a31_0
libmklml                  2019.0.5                      0
libpng                    1.6.37               h2a8f88b_0
libprotobuf               3.10.1               h7bd577a_0
libsodium                 1.0.16               h9d3ae62_0
lmu                       0.1                       dev_0    <develop>
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
markdown                  3.1.1                    py36_0
markupsafe                1.1.1            py36he774522_0
matplotlib                3.1.1            py36hc8f65d3_0
mistune                   0.8.4            py36he774522_0
mkl                       2019.4                      245
mkl-service               2.3.0            py36hb782905_0
mkl_fft                   1.0.15           py36h14836fe_0
mkl_random                1.1.0            py36h675688f_0
more-itertools            7.2.0                    py36_0
msys2-conda-epoch         20160418                      1
nbconvert                 5.6.1                    py36_0
nbformat                  4.4.0                    py36_0
nengo                     2.8.0                    pypi_0    pypi
nengolib                  0.5.2                    pypi_0    pypi
ninja                     1.9.0            py36h74a9793_0
notebook                  6.0.2                    py36_0
numpy                     1.17.4           py36h4320e6b_0
numpy-base                1.17.4           py36hc3f5095_0
openssl                   1.1.1d               he774522_3
opt_einsum                3.1.0                      py_0
pandas                    0.25.3           py36ha925a31_0
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py36_1
parso                     0.5.1                      py_0
patsy                     0.5.1                    py36_0
phased-lstm-keras         1.0.2                    pypi_0    pypi
pickleshare               0.7.5                    py36_0
pip                       19.3.1                   py36_0
prometheus_client         0.7.1                      py_0
prompt_toolkit            3.0.2                      py_0
protobuf                  3.10.1           py36h33f27b4_0
pycparser                 2.19                     py36_0
pygments                  2.5.2                      py_0
pyparsing                 2.4.5                      py_0
pyqt                      5.9.2            py36h6538335_2
pyreadline                2.1                      py36_1
pyrsistent                0.15.6           py36he774522_0
python                    3.6.9                h5500b2f_0
python-dateutil           2.8.1                      py_0
pytorch                   1.0.1           cpu_py36h39a92a0_0
pytz                      2019.3                     py_0
pywin32                   223              py36hfa6e2cd_1
pywinpty                  0.5.5                 py36_1000
pyyaml                    5.1.2            py36he774522_0
pyzmq                     18.1.0           py36ha925a31_0
qt                        5.9.7            vc14h73c81de_0
scikit-learn              0.21.3           py36h6288b17_0
scipy                     1.3.2            py36h29ff71c_0
seaborn                   0.9.0              pyh91ea838_1
send2trash                1.5.0                    py36_0
setuptools                42.0.2                   py36_0
sip                       4.19.8           py36h6538335_0
six                       1.13.0                   py36_0
sqlite                    3.30.1               he774522_0
statsmodels               0.10.1           py36h8c2d366_0
tensorboard               2.0.0              pyhb38c66f_1
tensorflow                1.15.0          mkl_py36h997801b_0
tensorflow-base           1.15.0          mkl_py36h190a33d_0
tensorflow-estimator      1.15.1             pyh2649769_0
termcolor                 1.1.0                    py36_1
terminado                 0.8.3                    py36_0
testpath                  0.4.4                      py_0
tornado                   6.0.3            py36he774522_0
traitlets                 4.3.3                    py36_0
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.16.27012          hf0eaf9b_1
wcwidth                   0.1.7                    py36_0
webencodings              0.5.1                    py36_1
werkzeug                  0.16.0                     py_0
wheel                     0.33.6                   py36_0
wincertstore              0.2              py36h7fe50ca_0
winpty                    0.4.3                         4
wrapt                     1.11.2           py36he774522_0
yaml                      0.1.7                hc54c509_2
zeromq                    4.3.1                h33f27b4_3
zipp                      0.6.0                      py_0
zlib                      1.2.11               h62dcd97_3
```
the output from `pip list`:
```
Package              Version             Location
-------------------- ------------------- -------------------------------------
absl-py              0.8.1
astor                0.8.0
attrs                19.3.0
backcall             0.1.0
bleach               3.1.0
certifi              2019.11.28
cffi                 1.13.2
colorama             0.4.1
cycler               0.10.0
decorator            4.4.1
defusedxml           0.6.0
entrypoints          0.3
gast                 0.2.2
google-pasta         0.1.8
grpcio               1.16.1
h5py                 2.9.0
importlib-metadata   1.2.0
ipykernel            5.1.3
ipython              7.10.1
ipython-genutils     0.2.0
jedi                 0.15.1
Jinja2               2.10.3
joblib               0.14.0
jsonschema           3.2.0
jupyter-client       5.3.4
jupyter-core         4.6.1
Keras                2.2.4
Keras-Applications   1.0.8
Keras-Preprocessing  1.1.0
kiwisolver           1.1.0
lmu                  0.1                 
Markdown             3.1.1
MarkupSafe           1.1.1
matplotlib           3.1.1
mistune              0.8.4
mkl-fft              1.0.15
mkl-random           1.1.0
mkl-service          2.3.0
more-itertools       7.2.0
nbconvert            5.6.1
nbformat             4.4.0
nengo                2.8.0
nengolib             0.5.2
notebook             6.0.2
numpy                1.17.4
opt-einsum           3.1.0
pandas               0.25.3
pandocfilters        1.4.2
parso                0.5.1
patsy                0.5.1
phased-lstm-keras    1.0.2
pickleshare          0.7.5
pip                  19.3.1
prometheus-client    0.7.1
prompt-toolkit       3.0.2
protobuf             3.10.1
pycparser            2.19
Pygments             2.5.2
pyparsing            2.4.5
pyreadline           2.1
pyrsistent           0.15.6
python-dateutil      2.8.1
pytz                 2019.3
pywin32              223
pywinpty             0.5.5
PyYAML               5.1.2
pyzmq                18.1.0
scikit-learn         0.21.3
scipy                1.3.2
seaborn              0.9.0
Send2Trash           1.5.0
setuptools           42.0.2.post20191203
six                  1.13.0
statsmodels          0.10.1
tensorboard          2.0.0
tensorflow           1.15.0
tensorflow-estimator 1.15.1
termcolor            1.1.0
terminado            0.8.3
testpath             0.4.4
torch                1.0.1
tornado              6.0.3
traitlets            4.3.3
wcwidth              0.1.7
webencodings         0.5.1
Werkzeug             0.16.0
wheel                0.33.6
wincertstore         0.2
wrapt                1.11.2
zipp                 0.6.0
```

**Comment by jabowery** [comment]
In order to get CUDA to work (but only with `accuracy` and `val_accuracy`) here is what I did:
```
conda create -n lmucondacuda python=3.6
conda activate lmucondacuda
conda install pytorch==1.2.0 torchvision==0.4.0 cudatoolkit=10.0 -c pytorch
cd lmu
```
Edit `setup.py` to remove `torch` from the `install_requires`:
```
      install_requires=[
          'keras>=2.2.4',
          'tensorflow>=1.12.0,<2',
          'scikit-learn>=0.20.1',
          'matplotlib>=3.0.2',
          'IPython>=7.2.0',
          'notebook>=5.7.4',
          'seaborn>=0.9.0',
          'numpy>=1.15.4',
          'scipy<1.3.0',
          'nengolib>=0.5.1',
          'nengo>=2.8.0,<3.0',
          'phased-lstm-keras',
      ],
```
Then run:
```
pip install -e .
```
Then (here's the real gymnastics) copy `Miniconda3\pkgs\cudnn-7.6.4-cuda10.0_0\Library\bin\cudnn64_7.dll` to `Miniconda3\envs\lmucondacuda\Library\bin`

**Comment by arvoelke** [comment]
This should be fixed by #18 which cleaned up the version dependencies, and has been released on pypi (`pip install lmu==0.1.0`; https://pypi.org/project/lmu/). Please feel free to create a new issue if there is any problem in `pip install lmu` on Python 3.5+.
